great_expectations/data_context/data_context/base_data_context.py

```diff
@@ -129,8 +131,8 @@
 yaml.indent(mapping=2, sequence=4, offset=2)
 yaml.default_flow_style = False
 
-
-class BaseDataContext(ConfigPeer):
+# TODO: <WILL> Most of the logic here will be migrated to EphemeralDataContext
```

Comment:
Thinking aloud... EphemeralDataContext is what we have today as BaseDataContext instantiated with DataContextConfig, the latter specified in Python, where stores may point to different store backends.  Does this work?  Or must EphemeralDataContext be such that even all aspects of its config are in memory?..

Reply:
As I understand it, it will only be the `DataContextConfig` that is specified in memory.  Within the configuration, DataContext would be able to point to any store backend that it would like.